### ISSUE TITLE
detail definition of SM and NH field in SAM format

### ISSUE THREAD

**Issue by xuan-wang** [samtools/hts-specs#187]
Hi,
In the SAM Optional Fields Specification [SAMtags.tex](http://samtools.github.io/hts-specs/SAMtags.pdf), there are fields termed as SM(Template-independent mapping quality) and NH(Number of reported alignments that contains the query in the current record).

BWA mem output sam doesn't contain them.

I am confused about the description and didn't find any detail definition about these two fields.
I need these two field to filter reads for SV calling. Could anybody tell me the meaning and how to generate them? 

Thanks a lot!

**Comment by colinhercus** [comment]
Hi Dicor,

We generate these in Novoalign.

NH will be output if a read has multiple mappings with similar alignment
score. It is the number of mappings found not the number of SAM lines for
the read.
The HI tag is the number of mappings actually appearing as SAM records for
the read.

SM we interpreted as MAPQ for the read if it wasn't part of a pair. For SVs
the read isn't mapped as a proper pair and the SAM MAPQ field applies to
each read independently. It's only in proper pairs that we calculate and
report SM.

Colin

**Comment by jmarshall** [comment]
The SM description was expanded to “Template-independent mapping quality, i.e., the mapping quality if the read were mapped as a single read rather than as part of a read pair or template” in dea1302f843dea1b754f7ba36db0b8858231c184.

The current NH description is “Number of reported alignments that contain the query in the current record”. Aided by the comment here, I'm planning to expand that along the lines of

> Number of reported alignments that contain the query in the current record. This is a total count of ‘acceptable’ alignments as identified by the aligner, and may be greater than the number of alignment records it actually emitted.

SAMtags currently describes HI as “Query hit index, indicating the alignment record is the i-th one stored in SAM” and IH as “Number of alignments stored in the file that contain the query in the current record”. (See also PR #326, which changed IH's description in passing and perhaps over-summarised its entry in the introductory table.) @colinhercus Is HI in your comment perhaps a typo? — IH instead would be consistent with the spec's descriptions.

**Comment by colinhercus** [comment]
@jmarshall  You're right, I should have said IH is the number of alignments we reported and IH the index of the alignment. I believe we follow spec's as they were in 2009.

With regard your new definition for NH “Number of reported alignments" I find the use of 'reported' as confusing. For me 'reported alignments' would be  mean alignments appearing in the report. 
Perhaps "Number of acceptable alignments identified by the aligner ....  Where 'acceptable' is defined by the aligner."

**Comment by jkbonfield** [comment]
Agreed on "reported" not being the ideal word, as the output is essentially the report from the aligner.  I'll also offer "identified" or "detected" as suitable alternatives; probably the latter as identified is used in the next sentence.

**Comment by jmarshall** [comment]
”Reported” has been there since 2009 but indeed if the word should be used at all, it should be in describing IH. Okay, I'll revisit that sentence too; PR incoming in due course…

(I'm presuming you meant “and **HI** the index of the alignment” :smile:)

**Comment by colinhercus** [comment]
Totally confused myself, transposition errors are always a problem for one finger typists :)
